### DISCUSSION TITLE
UI Related Discussions 📫

### DISCUSSION THREAD

**Discussion by nomandhoni-cs** [bowtie-json-schema/bowtie#259] [Q&A] [answered]
Hey @Julian,
Are there any specific reasons to use the Horizontal line to separate both row? 
![Screenshot_139](https://github.com/bowtie-json-schema/bowtie/assets/92979541/38f049cb-7de0-4282-950b-a54b7aa1c97d)

**Comment by nomandhoni-cs** [comment]
Why don't we just use like this?
![Screenshot_140](https://github.com/bowtie-json-schema/bowtie/assets/92979541/c038b9df-643e-4413-a132-4ef21774eb47)

**Comment by Julian** [comment, marked as answer]
The top table is present in every Bowtie report (the `Dialect` and `Run` lines), the bottom one (which just includes the Commit line) comes from "run metadata" which may not be present in any report, or may contain extra entries, etc. -- in other words the rows are not "the same" because the bottom row is a custom parameter that was passed through (by `bowtie suite`) rather than one always present -- so I just wanted some way of visually separating the two. Another way would have been writing "Run Metadata" somewhere between the two I suppose.

The relevant commit is e35e7a98547cad1ed82dba3573a1b2bbeead9199 from yesterday -- where the idea of the functionality (from #26) is e.g. to pass through extra data for a specific run, but which has no specific meaning to Bowtie in general, and here it's being used to pass through the test suite commit.

The dialect info and date info on the other hand are things every single report will have.

But yeah improvements are certainly welcome, as I say I think we should delineate somehow but that doesn't necessarily need to be `<hr>` that was just the simplest.

**Reply by Julian** [reply]
As a more general example, in the future someone probably will want to run` bowtie run --metadata metadata.json ...` where `metadata.json` is some file containing e.g.:

```json
{
    "My Property": 27,
    "Homepage": "http://example.com"
}
```

and the functionality was meant to include those bits of data in the outputted HTML. (And the delimiter is to make sure it's easy to see which data came from Bowtie and which came from the user running Bowtie).

**Reply by nomandhoni-cs** [reply]
> As a more general example, in the future someone probably will want to run` bowtie run --metadata metadata.json ...` where `metadata.json` is some file containing e.g.:
> 
> ```json
> {
>     "My Property": 27,
>     "Homepage": "http://example.com"
> }
> ```
> 
> and the functionality was meant to include those bits of data in the outputted HTML. (And the delimiter is to make sure it's easy to see which data came from Bowtie and which came from the user running Bowtie).

Ohh,,, Maybe then we can use a seperate table for that to display metadatas.

**Reply by Julian** [reply]
Right, so it's currently already a separate `<table>` element, the `<hr`> is just because otherwise it's not obvious visually that that's the case. But if you have a suggestion I'm all ears!

**Reply by nomandhoni-cs** [reply]
> Right, so it's currently already a separate `<table>` element, the `<hr`> is just because otherwise it's not obvious visually that that's the case. But if you have a suggestion I'm all ears!

Yup, I had a look on the code. I saw both are in the seperate `<table>`,
In this question I highlighted the `<hr>` because it is only visible not the table. Table is visible as separate row.
